pkg/resources/grant_helpers.go

```diff
@@ -152,45 +147,29 @@ func readGenericGrant(data *schema.ResourceData, meta interface{}, builder snowf
 	}
 	priv := data.Get("privilege").(string)
 
-	rolesIn, sharesIn := expandRolesAndShares(data)
-
 	var roles, shares []string
-	d("foo")
 	for _, grant := range grants {
 		// Skip if wrong privilege
 		if grant.Privilege != priv {
 			continue
 		}
-		d(grant)
 		switch grant.GranteeType {
 		case "ROLE":
-			if !stringInSlice(grant.GranteeName, rolesIn) {
-				continue
-			}
 			roles = append(roles, grant.GranteeName)
 		case "SHARE":
-			// Shares get the account appended to their name, remove this
-			granteeNameStrippedAccount := StripAccountFromName(grant.GranteeName)
-			if !stringInSlice(granteeNameStrippedAccount, sharesIn) {
-				continue
-			}
-
 			shares = append(shares, grant.GranteeName)
```

Comment:
```suggestion
			shares = append(shares, granteeNameStrippedAccount)
```
I think that was one of the fixes from before. Maybe keep that line and the comment. but just remove the `if !stringInSlice.....`